application/src/main/java/org/eclipse/hono/application/ApplicationConfig.java

```diff
@@ -33,6 +33,17 @@
     private static final Vertx vertx = Vertx.vertx();
 
     /**
+     * Returns the HonoApplication instance responsible for initializing the
+     * Hono server components.
+     * 
+     * @return HonoApplication instance
+     */
+    @Bean
+    public HonoApplication honoApplication() {
```

Comment:
Why doesn't `Application` simply extend `HonoApplication`? Isn't that the pattern you had in mind with moving the common stuff to the `server` module?